WIP Create friends ACL and get rid of ACCESS_FRIENDS

jdalsem opened the pull request:
Fixes #3391, #5038
Refs #8073, #3872

jdalsem committed:
deprecate(breadcrumbs): breadcrumbs now use href instead of link

Fixes #10345

jdalsem committed:
feat(navigation): breadcrumbs is now a menu

jdalsem committed:
wip friends acl

jdalsem committed:
wip friends acl

jdalsem committed:
more acl work

jdalsem committed:
more acl work

jeabakker commented on engine/classes/ElggEntity.php:
```diff
@@ -897,6 +897,28 @@ public function getOwnedAccessCollections($options = []) {
+	 * @since 3.0
+	 */
+	public function getOwnedAccessCollection($subtype) {
+		if (!is_string($subtype)) {
```
`|| $subtype === ''`

jeabakker commented on engine/classes/ElggEntity.php:
```diff
@@ -1301,7 +1323,11 @@ protected function create() {
 
 		if ($access_id == ACCESS_DEFAULT) {
-			throw new \InvalidParameterException('ACCESS_DEFAULT is not a valid access level. See its documentation in elgglib.h');
+			throw new \InvalidParameterException('ACCESS_DEFAULT is not a valid access level. See its documentation in elgglib.php');
```
not `elgglib.php` but in `constants.php` ....

jeabakker commented on engine/lib/access.php:
```diff
@@ -506,6 +506,106 @@ function access_init() {
+ * @param \Elgg\Event $event event
+ *
+ * @return void
+ */
```
`@since` and `@internal`

jeabakker commented on engine/lib/access.php:
```diff
@@ -531,5 +631,11 @@ function access_test($hook, $type, $value, $params) {
 	$events->registerHandler('ready', 'system', 'access_init');
 
+	// friends ACL events
+	$events->registerHandler('create', 'user', 'access_friends_acl_create');
```
check if `create:after` is available

jdalsem replied:
not available

jeabakker committed:
chore(seeding): process faker image from local file

jeabakker committed:
chore(tests): report code coverage to scrutinizer

jeabakker committed:
chore(travis): prevent composer timeout during seeding

Using composer to call the (un)seeding script can cause timeouts,
calling the script direclty avoids this issue.

jdalsem committed:
Merge pull request #11394 from jeabakker/seeder-image-processing-2

chore(seeding): process faker image from local file

jeabakker committed:
fix(i18n): fallback to site language before English

fixes #4389

jeabakker committed:
chore(core): support parent function arguments in ElggSite::delete

ref #11399

jeabakker committed:
feat(discussion): replies have been moved to comments

fixes #10217, #8458, #11424, #9762

jdalsem committed:
Merge pull request #11436 from jeabakker/site-delete-23

chore(core): support parent function arguments in ElggSite::delete

jdalsem committed:
fixed comments

jdalsem committed:
fixed comments

jdalsem committed:
function docs

jdalsem committed:
function docs

jeabakker committed:
Merge remote-tracking branch 'remote/2.3' into upmerge-23

Conflicts:
	.travis.yml
	composer.json
	engine/classes/Elgg/Database/Seeds/Seed.php
	engine/classes/ElggSite.php

jdalsem committed:
chore(groups): fixed dropdown content for owner transfer

fixes #11383

jeabakker committed:
Merge pull request #11438 from jeabakker/upmerge-23

Upmerge 2.3 into master

jeabakker committed:
chore(developers): restore entity_explorer menu item

fixes #11441

jeabakker committed:
chore(developers): replace deprecated function usage

jeabakker committed:
feat(forms): form_vars and body_vars are now available in the footer

fixes #11398

jdalsem committed:
Merge pull request #11425 from jeabakker/language-fallback

fix(i18n): fallback to site language before English

jdalsem committed:
Merge pull request #11439 from jdalsem/group-dropdown

chore(groups): fixed dropdown content for owner transfer

jdalsem committed:
Merge pull request #11431 from jeabakker/discussion-reply-to-comment

feat(discussion): replies have been moved to comments

jdalsem committed:
Merge pull request #11442 from jeabakker/developers-entity-explorer-menu

chore(developers): restore entity_explorer menu item

jdalsem committed:
Merge pull request #11443 from jeabakker/form-footer-vars

feat(forms): form_vars and body_vars are now available in the footer

jeabakker committed:
fix(tests): moved incorrect registered test to correct location

jeabakker committed:
fix(tests): incorrect filename for test registration

jeabakker committed:
fix(tests): correct registration of view path

jeabakker committed:
fix(tests): correct validation of action path

jdalsem committed:
feat(views): object/elements/imprint/time now support a href on the time

jdalsem committed:
feat(comments): submitting comments is now ajaxed

fixes #8133
fixes #8068

jdalsem committed:
Merge pull request #11448 from jeabakker/test-fixes

Test fixes

jdalsem committed:
Merge pull request #11101 from jdalsem/breadcrumb-href

Breadcrumb href

jeabakker committed:
Merge remote-tracking branch 'remote/2.3' into upmerge-23-20171205

Conflicts:
	engine/tests/phpunit/Elgg/ActionsServiceTest.php
	engine/tests/phpunit/Elgg/Database/PluginsTest.php
	engine/tests/phpunit/Elgg/ViewsServiceTest.php

jdalsem committed:
Merge pull request #11451 from jeabakker/upmerge-23-20171205

Upmerge 2.3 to master

jeabakker committed:
chore(release): v2.3.5

jdalsem committed:
Merge pull request #11452 from jeabakker/release-2.3.5

chore(release): v2.3.5

jdalsem committed:
feat(ckeditor): updated to full ckeditor v4.7.3 using composer

fixes #10968
fixes #9283

jeabakker committed:
Merge remote-tracking branch 'remote/2.3' into upmerge-2.3.5

Conflicts:
	composer.json
	composer.lock
	docs/locale/pot/contribute.pot

jdalsem committed:
Merge pull request #11454 from jeabakker/upmerge-2.3.5

Upmerge 2.3.5

jeabakker committed:
fix(navigation): correct container set on title menu button

When viewing another users content listing page the title menu button
'add content' could point to the wrong container, namely the listed user
instead of the current logged in user.
Now the button points to the current logged in user

fixes: #10846

jdalsem committed:
feat(members): use filter menu and resource views for the pages

fixes #10313

jdalsem committed:
feat(members): added members tabs to member search page

fixes #11298

jdalsem committed:
chore(users): correctly update last_action for ElggUser

fixes #11306

jdalsem committed:
Merge pull request #11455 from jeabakker/register-add-title-button2

fix(navigation): correct container set on title menu button

jdalsem committed:
Merge pull request #11453 from jdalsem/ckeditor5

feat(ckeditor): updated to full ckeditor v4.7.3 using composer

jeabakker committed:
fix(cron): log correct completed time in cron monitor

fixes #10797

jdalsem committed:
Merge pull request #11457 from jdalsem/member-fixes

Member fixes

jdalsem committed:
feat(likes): like annotation access is no longer linked to entity access

fixes #10981

jeabakker committed:
fix(cron): cron/run endpoint calls intervals reliably

fixes #8918
fixes #10803

hypeJunction committed:
feat(search): adds a new core search service

Adds a new search service that behaves similar to elgg_get_entities()

BREAKING CHANGE
Most of search plugin views have been removed or updated
Tag search has been combined with entity search.
Some of the search behavior may have changed, search hooks have been deprecated.

Fixes #7392
Fixed #11330
Fixes #11274
Fixes #7062
Fixed #3549

jdalsem committed:
Merge pull request #11449 from jdalsem/ajax-comments

Ajax comments

jdalsem committed:
Merge pull request #11462 from jdalsem/likes-access

feat(likes): like annotation access is no longer linked to entity access

jdalsem committed:
Merge pull request #11463 from jeabakker/cron-misc

Cron fixes

jdalsem committed:
feat(core): elgg_get_config supports a default value if config not set

jeabakker committed:
Merge pull request #11447 from hypeJunction/search

feat(search): adds a new core search service

jdalsem committed:
feat(views): friendly time switches to date format after a few days

The amount of days can be configured by the site admin.

Fixes #9897

jdalsem committed:
chore(i18n): plugin translations have been cleaned

fixes #10261

jdalsem committed:
chore(groups): removed old upgrade file

fixes #11461

jdalsem committed:
chore(search): cleanup of some issues after big refactor

jdalsem committed:
chore(ckeditor): removed some dependent plugins in config

jdalsem committed:
feat(tests): travis now also test on PHP 7.2

fixes #11444

hypeJunction committed:
feat(theme): new theme

Implements a new theme

BREAKING CHANGE
Most of the CSS has been rewritten, which will probably affect plugins
and themes. Plugins will need to adapt to new styling guidelines.

Fixes #11134
Fixes #10201
Fixes #7658
Fixes #10857
Fixes #10316
Fixes #4762
Fixes #11245
Fixes #6912

mrclay committed:
chore(sql): removes reverse_order_by query option

All usages have been replaced by explicit `order_by` values.

Fixes #11460

BREAKING CHANGE:
`reverse_order_by` in $options is ignored.

hypeJunction committed:
chore(messages): unread count works again

Fixes #11476

jdalsem committed:
Merge pull request #11480 from mrclay/11460_reverse

chore(sql): removes reverse_order_by query option

jdalsem committed:
Merge pull request #11473 from jdalsem/small-master-fixes

Small master fixes

jdalsem committed:
Merge pull request #11466 from jdalsem/friendly-time

Friendly time

jdalsem committed:
Merge pull request #11472 from jdalsem/groups-upgrade

chore(groups): removed old upgrade file

mrclay committed:
fix(views): clean up of class attribute usage

Fixes #11468

jdalsem committed:
Merge pull request #11481 from mrclay/class_list

fix(views): clean up of class attribute usage

jdalsem committed:
Merge pull request #11474 from jdalsem/travis-php-7-2

feat(tests): travis now also test on PHP 7.2

jdalsem committed:
Merge pull request #11467 from jdalsem/plugin-translations

chore(i18n): plugin translations have been cleaned

jdalsem committed:
chore(search): updated usage of elgg_search in plugins

fixes #8724

jdalsem committed:
Merge pull request #11484 from jdalsem/plugin-search

chore(search): updated usage of elgg_search in plugins

jdalsem committed:
Merge pull request #11395 from jeabakker/code-coverage

chore(tests): report code coverage to scrutinizer

hypeJunction committed:
Merge branch 'master' into css

jdalsem committed:
Merge pull request #11477 from hypeJunction/css

feat(theme): improve the design

hypeJunction committed:
chore(theme): fix widget layouts

hypeJunction committed:
chore(theme): bring back admin layout header

jdalsem committed:
Merge pull request #11488 from hypeJunction/widgets

Theme fixes

jdalsem committed:
Merge branch 'friends_acl_last_time' of git@github.com:jdalsem/Elgg.git
into friends_acl_last_time

Conflicts:
	docs/appendix/upgrade-notes/2.x-to-3.0.rst

jdalsem commented:
grr... going to redo this
